flake.nix

```
@@ -1,22 +1,33 @@
 {
-  inputs = {
+  inputs = rec {
     crane.url = "github:ipetkov/crane";
     utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = github:NixOS/nixpkgs/nixpkgs-unstable;
+    crane.inputs.nixpkgs.follows = "nixpkgs";
   };
 
   outputs = {
     self,
     crane,
     utils,
+    nixpkgs,
     ...
   }: let
     supportedSystems = ["x86_64-linux" "x86_64-darwin" "aarch64-linux"];
   in
     utils.lib.eachSystem supportedSystems
     (
-      system: {
+      system:
+        let
+          pkgs = import nixpkgs { inherit system; };
+        in
+        {
         packages.oura = crane.lib.${system}.buildPackage {
           src = self;
+          cargoExtraArgs = "--features \"kafkasink\"";
+          nativeBuildInputs = with pkgs; [ openssl openssl.dev pkg-config ];
+          PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig";
+          RUST_BACKTRACE = "full";
```

```suggestion
      system: {
        packages.oura = crane.lib.${system}.buildPackage {
          src = self;
          cargoExtraArgs = "--features kafkasink";
          nativeBuildInputs = with import nixpkgs { inherit system; }; [ perl ];
```
no need for (escaped) quotes in `features`. as for perl, kafkasink uses openssl vendored, this has a build time dependency on perl. It's what they specify [in the docs](https://docs.rs/openssl/latest/openssl/#vendored). the rest is covered by stdenv.

This won't build on a pure machine without perl I think, your preinstalled perl might've made this not apparent, just a guess!

---

Yeah, this is Perl, of course! Good catch!